reqs/srd/SRD004.yml

```diff
@@ -0,0 +1,17 @@
+active: true
+derived: false
+header: |
+  Contract enforcement level
+level: 1.4
+links: []
+normative: true
+ref: ''
+references:
+- keyword: SRD004
+  path: include/contracts_lite/enforcement.hpp
+  type: file
+reviewed: lxvURN6_ps2Z9N9zVSXTDpBWqa1E6hSmVGhAhmadh9c=
+text: |
+  The contracts library _shall_ provide `DEFAULT`, `AUDIT`, and `OFF` enforcement mechanisms.
+
+  The `DEFAULT` enforcement level is intended to be used for contract enforcement that is intended to always run. The `AUDIT` level is intended for checks that are only intended to run offline or during some kind of system testing or audit (these tend to be resource intensive in ways that prohibit them from running during normal operation). The `OFF` enforcement level removes all checks from the compiled code.
```

Comment:
I heard feedback from a Functional Safety person that took issue with calling it an `AUDIT` level since it reminded him of Dieselgate (i.e., when an auditor turns up, enable the "right" checks, but during normal operation, disable those). AUDIT stems from the C++ contracts standard but maybe we should soften the _"or during some kind of system testing or audit"_ to remove the audit bits.. (or is this also taken from the language in the C++ contracts standard?). Essentially we should be clear that audit means "additional testing" rather than enabling something to convince a safety auditor on site, versus hiding something at runtime..

Reply:
That's interesting, and a good point. I opened an issue to look at that: #11

Reply:
discussed offline: some clarifying text will be added to the readme and to the requirements yaml in a separate PR.